wordpress-mobile/WordPress-iOS
# Feature/highlights tooltip presenter

Refs pbArwn-4oo-p2

This PR adds `TooltipPresenter` which calculates where to place the Tooltip in respect to another view. One only needs to provide the `containerView` (most commonly `viewController.view`)  which contains the `targetView` and will contain the `Tooltip`. The `targetView` is where the tooltip will point at and `Tooltip` should be instantiated & configured before it is passed in.

The tooltip & the arrow will be placed to point at the `midX` of the `targetView`.

To test:

- Please first place the following code in the `viewDidAppear` of `ReaderDetailViewController`.
- Then select a post from "Reader" and the tooltip should appear with a `dummySquare` which you can change its frame to see how the tooltip reacts.

```Swift
        // FIXME: Remove test code!
        let dummySquare = UIView()
        dummySquare.backgroundColor = .red
        dummySquare.translatesAutoresizingMaskIntoConstraints = false
        dummySquare.layer.cornerRadius = 4
        view.addSubview(dummySquare)

        NSLayoutConstraint.activate([
            dummySquare.leadingAnchor.constraint(equalTo: view.leadingAnchor, constant: 180),
            dummySquare.heightAnchor.constraint(equalToConstant: 30),
            dummySquare.widthAnchor.constraint(equalToConstant: 30),
            dummySquare.topAnchor.constraint(equalTo: view.topAnchor, constant: 120)
        ])

        view.layoutIfNeeded()

        let tooltip = Tooltip()

        tooltip.title = "Follow the conversation"
        tooltip.message = "Get notified."
        tooltip.primaryButtonTitle = "Got it"
        tooltip.secondaryButtonTitle = "Learn more"

        tooltipPresenter = TooltipPresenter(
            containerView: view,
            tooltip: tooltip,
            targetView: dummySquare
        )
        tooltipPresenter?.show()
```
Also retain the `tooltipPresenter` to respond to orientation and system font size changes:
```Swift 
private var tooltipPresenter: TooltipPresenter?
```

## Regression Notes
1. Potential unintended areas of impact
N/A. Isolated addition that has no impact on rest of the code.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

3. What automated tests I added (or what prevented me from doing so)
N/A. It is purely UI logic.

PR submission checklist:


## Screenshots

| Full Width  | Narrow |
| ------------- | ------------- |
| ![Screenshot on 2022-05-13 at 10:48:14](https://user-images.githubusercontent.com/15968946/168276644-dcbbf7dd-8cbe-42aa-a364-ce9ee47c808c.png)  | ![Screenshot on 2022-05-13 at 11:08:55](https://user-images.githubusercontent.com/15968946/168276783-502a927c-2041-40f8-83c5-a497bca081d1.png) |
| ![Screenshot on 2022-05-13 at 10:52:04](https://user-images.githubusercontent.com/15968946/168276684-6bca642e-2039-4893-92a4-efb3bd07877e.png) | ![Screenshot on 2022-05-13 at 11:03:06](https://user-images.githubusercontent.com/15968946/168276757-44e2e4ec-2f08-44c0-8e05-165eadc171be.png)  |
